pkg/runtime/wfengine/wfengine.go

```diff
@@ -46,38 +43,61 @@ type WorkflowEngine struct {
 	actorRuntime   actors.Actors
 	startMutex     sync.Mutex
 	disconnectChan chan any
+	config         *WFConfig
 }
 
 var (
 	wfLogger            = logger.NewLogger("dapr.runtime.wfengine")
 	errExecutionAborted = errors.New("execution aborted")
 )
 
+// WFConfig is the configuration for the workflow engine
+type WFConfig struct {
+	AppID             string
+	WorkflowActorType string
+	ActivityActorType string
+}
+
 func IsWorkflowRequest(path string) bool {
 	return backend.IsDurableTaskGrpcRequest(path)
 }
 
-func NewWorkflowEngine() *WorkflowEngine {
+func NewWorkflowEngine(config *WFConfig) *WorkflowEngine {
+	actorTypePerAppPrefix := strings.Builder{}
+	actorTypePerAppPrefix.WriteString(actors.InternalActorTypePrefix)
+	actorTypePerAppPrefix.WriteString(utils.GetNamespaceOrDefault())
+	actorTypePerAppPrefix.WriteString(utils.DotDelimiter)
+	actorTypePerAppPrefix.WriteString(config.AppID)
+	actorTypePerAppPrefix.WriteString(utils.DotDelimiter)
+	actorTypePerAppPrefixStr := actorTypePerAppPrefix.String()
+
+	config.WorkflowActorType = actorTypePerAppPrefixStr + utils.WorkflowNameLabelKey
+	config.ActivityActorType = actorTypePerAppPrefixStr + utils.ActivityNameLabelKey
```

Comment:
It doesn't make sense to initialize the config object that's passed in as a constructor. I think of this as a bit of an anti-pattern since config objects should generally be immutable. It seems to me that you should simply pass compute the actor type value using the app ID and namespace and then just pass the actor type value into the `NewActivityActor` and `NewWorkflowActor` functions.

Reply:
Alternatively, consider initializing the workflow and activity actor types when constructing the `WFConfig` object (perhaps using a `NewWorkflowConfig` function) and exposing those types via read-only methods or fields.

Reply:
Changed as discussed